packages/core/src/wallet-manager.ts

```diff
@@ -0,0 +1,249 @@
+  }
+
+  get address() {
+    return this.data?.address as string;
+  }
+
+  get activeWallets() {
+    return this.walletRepo.activeItems;
+  }
+
+  get walletNames() {
+    return this.activeWallets.map((item) => item.name);
+  }
+
+  get walletCount() {
+    return this.walletNames.length;
+  }
+
+  get activeChains() {
+    return this.chainRepo.activeItems;
+  }
+
+  get chainNames() {
+    return this.activeChains.map((item) => item.name);
+  }
+
+  get chainCount() {
+    return this.chainNames.length;
+  }
+
+  get concurrency() {
+    return this._concurrency;
+  }
+
+  emit(type: string) {
+    return this.actions?.[type];
+  }
+
+  get emitWalletName() {
+    return this.actions?.walletName;
+  }
+
+  get emitChainName() {
+    return this.actions?.chainName;
+  }
+
+  get emitModalOpen() {
+    return this.actions?.modalOpen;
+  }
+
+  setAction(actions: Actions) {
+    this.actions = actions;
+  }
+
+  setAutos(autos: Autos) {
+    this.autos = autos;
+  }
+
+  reset() {
+    this.currentWallet?.reset();
+  }
+
+  setCurrentWallet = (walletName?: WalletName) => {
+    this.reset();
+    this._currentWalletName = walletName;
+    this.emitWalletName?.(walletName);
+  }
+
+  setCurrentChain = (chainName?: ChainName) => {
+    this.reset();
+    this._currentChainName = chainName;
+    this.emitChainName?.(chainName);
+  }
+
+  useWallets = (walletNameInfo: WalletName[] | WalletName) => {
+    this.walletRepo.inactivateAll();
+    if (Array.isArray(walletNameInfo)) {
+      if (walletNameInfo.length === 0) {
+        throw new Error('No wallet provided!');
+      }
+      walletNameInfo.forEach((name) => {
+        this.walletRepo.activate(name);
+      });
+      this._useModal = true;
+    } else {
+      this.walletRepo.activate(walletNameInfo);
+      this.setCurrentWallet(walletNameInfo);
+      this._useModal = false;
+    }
+    console.info(`${this.walletCount} wallets are used!`)
+  }
+
+  useChains = (chainNames: ChainName[]) => {
+    this.chainRepo.inactivateAll();
+    chainNames.forEach((name) => {
+      this.chainRepo.activate(name);
+    });
+    this.walletRepo.registeredItemMap.forEach((item) => {
+      item.wallet.setSupportedChains(this.chainRepo.activeItems);
+    });
+    console.info(`${this.chainCount} chains are used!`)
+  }
+
+  private getWallet(walletName?: WalletName, chainName?: ChainName): ExtendedWallet | undefined {
+    if (!walletName) {
+      return undefined;
+    }
+
+    let wallet = this.walletRepo.getItem(walletName).wallet;
+    if (chainName) {
+      wallet = wallet.getChain(chainName);
+    }
+    wallet.actions = this.actions;
+    return wallet;
+  }
+
+  update = () => { }
+
+  connect = async () => {
+    if (!this.currentWalletName) {
+      this.openModal();
+      return
+    }
+    try {
+      await this.currentWallet.connect();
+
+      if (
+        this.walletStatus === WalletStatus.Connected
+        && this.autos?.closeModalWhenWalletIsConnected
```

Comment:
since we'll also be doing `Dropdown` in addition to `Modal`, what if we rename the props to be something like `View` instead?

e.g. this would become `closeViewWhenWalletIsConnected`

Reply:
ok